### PR TITLE
build(client): Fix configs preventing incremental build

### PR DESCRIPTION
**Pull request by tylerbutler**
Fixes several problems causing builds to not be fully incremental.

1. data-object-base had no tests but included test-related tasks. Since no test output was generated, the task was never incremental. Those superfluous tasks have been removed. They can be added back if/when the package is no longer experimental.
2. Added new declarative tasks for `flub check buildversion` and `markdown-magic`.
3. The ai-collab project has `"allowJs": true` in its tsconfig. That setting causes problems with incremental builds (see #23197 for more details). It's removed in this change but if it is truly needed then I'll add it back.

### CODE REVIEW

File: fluidBuild.config.cjs [hunk 1]
```diff
@@ -160,10 +160,35 @@ module.exports = {
 
 	multiCommandExecutables: ["oclif", "syncpack"],
 	declarativeTasks: {
+		"flub check buildVersion": {
```
Comment (tylerbutler) [inline fluidBuild.config.cjs:163]: This unfortunately doesn't work - need to debug why.

File: examples/apps/ai-collab/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
 	"extends": "../../../common/build/build-common/tsconfig.node16.json",
 	"compilerOptions": {
 		"lib": ["dom", "dom.iterable", "esnext"],
-		"allowJs": true,
+		// "allowJs": true,
```
[the remark this answers is not in the document]
Comment (alexvy86) [reply examples/apps/ai-collab/tsconfig.json:5]: I don't think so. @seanimam was this put there deliberately?

File: fluidBuild.config.cjs [hunk 3]
```diff
@@ -160,10 +160,39 @@ module.exports = {
...
+		},
 		"jssm-viz": {
 			inputGlobs: ["src/**/*.fsl"],
 			outputGlobs: ["src/**/*.fsl.svg"],
 		},
+		"markdown-magic": {
+			inputGlobs: [],
+			outputGlobs: [
+				// release group packages; while ** is supported, it is very slow, so these entries capture all the levels we
+				// have packages at today. Once we can upgrade to a later version of
```
Comment (alexvy86) [inline fluidBuild.config.cjs:187]:
```suggestion
				// have generated markdown files at today. Once we can upgrade to a later version of
```

**Review by alexvy86 — APPROVED** [review]
Pending confirmation of the allowJs bit, this lgtm.

File: examples/apps/ai-collab/tsconfig.json (hunk 2 above)
Comment (tylerbutler) [inline examples/apps/ai-collab/tsconfig.json:5]:
```suggestion
```

**Pull request merged**